Makefile.am

```diff
@@ -415,6 +415,13 @@ uninstall-local:
 	-rm $(DESTDIR)$(udevrulesdir)/$(udevrulesprefix)tpm-udev.rules
 endif
 
+install-exec-hook:
+	cd $(DESTDIR)$(libdir) && \
+		$(LN_S) libtss2-tcti-device.so libtss2-tcti-default.so
+
+uninstall-hook:
+	unlink $(DESTDIR)$(libdir)/libtss2-tcti-default.so
```

Comment:
Note that despite the name, `unlink` will remove `libtss2-tcti-default.so` regardless of whether it is a symbolic link or not. In tpm2-tss-engine there is an additional sanity check in the form of [`[ -L file ] && rm -f file`](https://github.com/tpm2-software/tpm2-tss-engine/blob/d79620378b681ecc2b6b63745e61a95ebc9b804e/Makefile.am#L92) to avoid deleting real files. In general I think using `rm -f` instead of `unlink` is the more standard way to do this (but both are POSIX, so should be portable).

Reply:
"[ -L file ] && rm -f file" produces uninstall-hook Error if the file/link doesn't exist. I will just do rm -f file.

Reply:
Ah yes, if the file doesn't exist, that would bail out because the last command to be executed is `[ -L file ]`, which exits with a non-zero exit status. You can do
```
cd $(DESTDIR)$(libdir) && [ -L libtss2-tcti-default.so ] && rm -f libtss2-tcti-default.so || true
```
to continue uninstalling even if the file is not found.